### PR TITLE
feat: plug tabular api to udata-tabular-preview (decommission of csvapi)

### LINKED ISSUE #1309: Utilisation d'api-tabular plutôt que csvapi dans la préviz datagouv
A faire dans udata-tabular-preview

### PR DESCRIPTION
**Pull request by geoffreyaldebert**
This PR plugs new tabular-api to the plugin and decommission csvapi. There is no more dependency to repository explore.data.gouv.fr. All logic (quite simple) is stored in this plugin. Now we have a endpoint for getting rows and another for getting profile, previously there was only one endpoint for both.
We also review style of preview with in particular column width and cell height fixed and a nice banner for explore CTA. Exemple below : 

image
and
image

We do not trust mime type provided by the user. We only trust the hydras extras `analysis:parsing:finished_at` that told us if an existing table exist on postgres.

Data Structure panel display is bugged right now, [an issue](https://github.com/etalab/data.gouv.fr/issues/1304) exists and @nicolaskempf57 needs to update components `datagouv-components` and release udata to work.

Finally, url  of explore button is also changed to link with new display in explore.data.gouv.fr (the one plugged with tabular-api).

⚠️ Change of udata.cfg ENV : `TABULAR_CSVAPI_URL` to `TABULAR_API_URL`

⚠️ This provoke an impossibility of apify url that is not already processed by [hydra](https://github.com/etalab/udata-hydra/)

Closes https://github.com/etalab/data.gouv.fr/issues/1309

### CODE REVIEW

File: udata_tabular_preview/settings.py [hunk 1]
```diff
@@ -18,7 +18,5 @@
 
 # Supported mimes
 TABULAR_SUPPORTED_MIME_TYPES = (
-    'text/csv',
-    'application/vnd.ms-excel',
-    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'text/csv', "text/plain", "text/csv"
```
Comment (nicolaskempf57) [inline udata_tabular_preview/settings.py]: Is there a reason to have `text/csv` twice ?

File: udata-tabular-preview/explore.vue [hunk 2]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
 });
 </script>
+
+<style scoped>
+.style-cell{
+  height: 3rem;
+  overflow-y: auto;
+  display: block;
+  width: 100%;
+  font-size: 12px;
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
  font-size: 0.75rem;
```

File: udata-tabular-preview/explore.vue [hunk 3]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+  line-height: 20px;
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
  line-height: 1.25rem;
```

File: udata-tabular-preview/explore.vue [hunk 4]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+  }
+
+  .text-explore{
+    padding-left: 0.5rem;
+    padding-right: 0.5rem;
+    margin-right: 1rem;
+  }
+  .title-text-explore{
+    font-weight: bold;
+    font-size: 16px;
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
    font-size: 1rem;
```

File: udata-tabular-preview/explore.vue [hunk 5]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+    line-height: 24px;
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
    line-height: 1.5rem;
```

File: udata-tabular-preview/explore.vue [hunk 6]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+    color: #3558A2;
+  }
+  .subtitle-text-explore{
+    font-size: 14px;
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
    font-size: 0.875rem;
```

File: udata-tabular-preview/explore.vue [hunk 7]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+    line-height: 18px;
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
    line-height: 1.125rem;
```

File: udata-tabular-preview/explore.vue [hunk 8]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+td{
+  border-right: 1px solid #CECECE;
+  border-bottom: 1px solid #CECECE;
+}
+th{
+  border-right: 1px solid #CECECE;
+}
+
+
+@media (min-width: 40rem){
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
@media (min-width: 48em){
```

`48em` is the DSFR MD breakpoint

File: udata-tabular-preview/explore.vue [hunk 9]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+    font-style: italic;
+    color: #3558A2;
+  }
+
+  .button-explore{
+    margin-left: auto;
+  }
+}
+
+@media (max-width: 40rem){
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]:
```suggestion
@media (max-width: 48em){
```

File: udata-tabular-preview/explore.vue [hunk 10]
```diff
@@ -74,37 +96,116 @@ export default defineComponent({
...
+      currentPage,
+      sortConfig,
+      isSortedBy,
+      sortByField,
+      changePage,
+      getTableInfos,
+    } = getDataTabularApi(props.resource)
 
-    apifyAndGetData();
+    getTableInfos(currentPage.value);
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue:101]: Is `sortConfig.value` missing here as 2nd parameter ?

File: udata-tabular-preview/explore.vue [hunk 11]
```diff
@@ -1,67 +1,89 @@
...
+        <p class="fr-grid-row fr-m-0">
+          <span class="fr-icon-warning-line" aria-hidden="true"></span>
+          {{ $t("The preview of this file failed to load.") }}
+        </p>
       </div>
+      <Loader v-if="loading" />
+      <template v-if="!hasError && !loading">
+        <div class="banner-explore">
+          <div class="france-explore">
+            <svg width="80" height="80" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (nicolaskempf57) [inline udata-tabular-preview/explore.vue]: You can move this svg to its own `.svg` file. To make it accessible, an `aria-hidden="true"` attribute is required.
To use the new svg file here, you have to :
- import it in script `import franceSvg from "./france.svg?raw";`
- return franceSvg in the setup function
- add `v-html="franceSvg"` to the france-explore div

File: udata-tabular-preview/tabularapi.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    })
+    .then(data => {
+      return { data };
+    });
+}
+
+/**
+ * Call Tabular-api to get table content
+ * @param {string} id resource id
+ * @param {string} page page
```
Comment (nicolaskempf57) [inline udata-tabular-preview/tabularapi.js]:
```suggestion
 * @param {number} page page
```

**Review by nicolaskempf57 — COMMENTED** [review]
Some minor changes but it looks good to me !

**Comment by nicolaskempf57** [comment]
I think we can also remove the `csvapi.js` file and the dependency to `@etalab/explore.data.gouv.fr`

File: udata-tabular-preview/explore.vue (hunk 10 above)
Comment (geoffreyaldebert) [reply udata-tabular-preview/explore.vue:101]: for me sort is reset so we do not need this param.

Comment (nicolaskempf57) [reply udata-tabular-preview/explore.vue:101]: The `getTableInfos` doesn't have a default value for its 2nd parameter. If we don't provide one here, we should add a default value in `getTableInfos` declaration.

File: udata-tabular-preview/locales/fr.json [hunk 13]
```diff
@@ -1,12 +1,13 @@
...
   "The preview of this file failed to load.": "La prévisualisation de ce fichier n'a pas pu être chargée.",
   "{count} rows": "{count} lignes | {count} ligne | {count} lignes",
   "Sort ascending": "Trier par ordre croissant",
   "Sort descending": "Trier par ordre décroissant",
   "The data structure of this file failed to load.": "La structure de données de ce fichier n'a pas pu être chargée.",
-  "This is an Excel file, analysis on this type of files are limited.": "Il s'agit d'un fichier Excel, l'analyse sur ce type de fichiers est limitée.",
-  "No data structure found for this file.": "Aucune structure de données trouvée pour ce fichier."
+  "No data structure found for this file.": "Aucune structure de données trouvée pour ce fichier.",
+  "Explore data in detail": "Explorer les données en détail",
+  "Use our tool": "Utiliser notre outil pour obtenir un aperçu des données, en savoir plus sur les différentes colonnes ou réaliser des filtres et des tris."
```
Comment (maudetes) [inline udata-tabular-preview/locales/fr.json:12]: We usually do these translations in Crowdin. I'm okay applying these directly if it doesn't raise issue. What do you think @nicolaskempf57?

File: udata_tabular_preview/preview.py [hunk 14]
```diff
@@ -36,8 +34,7 @@ def can_preview(self, resource):
             not max_size
             or (extras_size or resource.filesize or float("inf")) <= max_size
         )
-
         return all((has_config, is_supported, is_allowed, size_ok))
 
     def preview_url(self, resource):
-        return f'{self.preview_base_url}/?url={quote_plus(resource.latest)}'
+        return f'{self.preview_base_url}/resources/{resource.id}'
```
Comment (maudetes) [inline udata_tabular_preview/preview.py:27]: Shouldn't it be `/datasets/{dataset.id}/resources/{resource.id}`?

File: udata_tabular_preview/preview.py [hunk 15]
```diff
@@ -12,8 +11,8 @@ def preview_base_url(self):
 
     def can_preview(self, resource):
```
Comment (maudetes) [inline udata_tabular_preview/preview.py:12]: Shouldn't we base `can_preview` on `analysis:parsing:finished_at` extras now following https://github.com/etalab/udata-hydra/pull/91?

File: udata_tabular_preview/__init__.py [hunk 16]
```diff
@@ -1,8 +1,8 @@
 '''
 Tabular Preview
 
-A plugin to preview tabular files (CSV, Excel, ...)
+A plugin to preview tabular CSV files
```
Comment (maudetes) [inline udata_tabular_preview/__init__.py]: We want to revert to CSV & Excel right? It has been modified at different places in code. See https://github.com/etalab/udata-hydra/pull/92

**Review by maudetes — COMMENTED** [review]
Nice! I like the new preview layout! :star_struck:

Almost here I think!

File: udata_tabular_preview/preview.py (hunk 14 above)
Comment (geoffreyaldebert) [reply udata_tabular_preview/preview.py:27]: it was difficult to get dataset so I developped another route in explore so no, this route is correct and will redirect to the right exploration. But I think error is due to a mistake in `udata__explore_url` param in deployment

File: udata_tabular_preview/__init__.py (hunk 16 above)
Comment (geoffreyaldebert) [reply udata_tabular_preview/__init__.py]: yes dev has been done before @Pierlou decided to preview excel file 😅

**Comment by geoffreyaldebert** [comment]
I don't understand why traduction "The preview of this file failed to load." does not work because there are present in locales... help @nicolaskempf57 
Cc @maudetes

File: udata_tabular_preview/preview.py (hunk 15 above)
Comment (geoffreyaldebert) [reply udata_tabular_preview/preview.py:12]: ok

**Comment by nicolaskempf57** [comment]
> I don't understand why traduction "The preview of this file failed to load." does not work because there are present in locales... help @nicolaskempf57 Cc @maudetes

This seems unrelated to this PR and already present on master and [demo](https://demo.data.gouv.fr/fr/datasets/bornes-de-recharges-arcachon/). The translations from the package are loaded but somehow erased during the initial setup.

**Comment by geoffreyaldebert** [comment]
ok thanks, so can you approve @maudetes if you're ok ?

**Review by maudetes — APPROVED** [review]
I think we're good, thank you! :clap: 

Don't forget the changelog.
Could you also please update the PR description and maybe remove the unnecessary settings (ex: TABULAR_SUPPORTED_MIME_TYPES)?

**Review by nicolaskempf57 — APPROVED** [review]

**Pull request merged**